apisix/admin/stream_routes.lua

```diff
@@ -25,23 +27,69 @@ local _M = {
     need_v3_filter = true,
 }
 
+local function check_router_refer(items, id)
+    local refer_list = {}
+    local referkey = "/stream_routes_refer/" .. id
+    local _, err = core.etcd.delete(referkey)
+    core.log.warn(err)
+    core.log.warn(items)
+    for _, item in config_util.iterate_values(items) do
+        if item.value == nil then
+            goto CONTINUE
+        end
+        local r_id = string.gsub(item["key"], "/", "_")
+        local route = item.value
+        if route.protocol and route.protocol.superior_id then
+            local data
+            local setkey = "/stream_routes_refer/" .. route.protocol.superior_id
+            local res, err = core.etcd.get(setkey, false)
```

Comment:
Do we have to filter the data by `superior_id` in etcd every time when we delete an item like this? That might be very inefficient.

Reply:
For  generate the reference-relationship while iterating  over the items(stream_routes),  only  the data by superior_id exists reference-relationship